amy/extrequests/forms.py

```diff
@@ -440,6 +453,31 @@ def institution_label_from_instance(obj):
         essentially works just like `Model.__str__`."""
         return "{}".format(obj.fullname)
 
+    @feature_flag_enabled("ENFORCE_MEMBER_CODES")
+    def clean_membership_info(self, request: HttpRequest) -> dict:
+        errors = dict()
+        affiliation = self.cleaned_data.get("member_affiliation")  # yes/no/unsure
+        code = self.cleaned_data.get("member_code", "")
+        if affiliation in ["yes", "unsure"] and code:
+            # ensure that code belongs to a membership
+            try:
+                Membership.objects.get(registration_code=code)
```

Comment:
I assume here you want to add other validation in future, for example like "is membership active"?

Reply:
Yes, this will come later #2533